src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp

```diff
@@ -262,12 +262,26 @@ namespace OpenMS
       // and all bytes except the least significant one will be zero. Thus
       // we can convert to char directly (only keeping the least
       // significant byte).
+      const XMLCh* it = chars;
+      const XMLCh* end = it + length;
+
       size_t curr_size = result.size();
       result.resize(curr_size + length);
-      for (size_t i = 0; i < length; i++)
-      {
-        result[curr_size + i] = (char)chars[i];
+      std::string::iterator str_it = result.begin();
+      std::advance(str_it, curr_size);
+      while (it!=end)
+      {   
+        *str_it = (char)*it;
+        ++str_it;
+        ++it;
       }
+
+      // This is ca. 50 % faster than 
```

Comment:
this is quite surprising. Migth this be because curr_size was not considered const by the compiler? If it is that way const size_t might have similar speed but is easier to read